template.yaml

```diff
@@ -522,35 +523,57 @@ Resources:
             BatchSize: 1
+
+  CloseZendeskTicketFunctionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Sub ${AWS::StackName}-${Environment}-close-zendesk-ticket-role
+      PermissionsBoundary:
+        !If [
+          UsePermissionsBoundary,
+          !Ref PermissionsBoundary,
+          !Ref AWS::NoValue
+        ]
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action: sts:AssumeRole
       Policies:
-        - Statement:
-            - Sid: Logs
-              Effect: Allow
-              Action:
-                - logs:CreateLogGroup
-                - logs:CreateLogStream
-                - logs:PutLogEvents
-              Resource: !GetAtt CloseZendeskTicketLogs.Arn
-            - Sid: ReadSecrets
-              Effect: Allow
-              Action:
-                - secretsmanager:GetSecretValue
-              Resource:
-                - '{{resolve:ssm:ZendeskSecretSetArn}}'
-            - Sid: AllowCloseZendeskTicketQueueRead
-              Effect: Allow
-              Action:
-                - sqs:ReceiveMessage
-                - sqs:DeleteMessage
-                - sqs:GetQueueAttributes
-              Resource: !Sub 'arn:aws:sqs:${AWS::Region}:{{resolve:ssm:QueryResultsAccountNumber}}:txma-qr-infra-${Environment}-close-zendesk-ticket-queue'
-            - Sid: DecryptKmsKeys
-              Effect: Allow
-              Action:
-                - kms:Decrypt
-              Resource:
-                - '{{resolve:ssm:SecretsKmsKeyArn}}'
-                - !Sub 'arn:aws:kms:${AWS::Region}:{{resolve:ssm:QueryResultsAccountNumber}}:*'
+        - PolicyName: AllowCloseZendeskActions
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Sid: Logs
+                Effect: Allow
+                Action:
+                  - logs:CreateLogGroup
+                  - logs:CreateLogStream
+                  - logs:PutLogEvents
+                Resource: !GetAtt CloseZendeskTicketLogs.Arn
+              - Sid: ReadSecrets
+                Effect: Allow
+                Action:
+                  - secretsmanager:GetSecretValue
+                Resource:
+                  - '{{resolve:ssm:ZendeskSecretSetArn}}'
+              - Sid: AllowCloseZendeskTicketQueueRead
+                Effect: Allow
+                Action:
+                  - sqs:ReceiveMessage
+                  - sqs:DeleteMessage
+                  - sqs:GetQueueAttributes
+                Resource: !Sub 'arn:aws:sqs:${AWS::Region}:{{resolve:ssm:QueryResultsAccountNumber}}:txma-qr-infra-${Environment}-close-zendesk-ticket-queue'
+              - Sid: DecryptKmsKeys
+                Effect: Allow
+                Action:
+                  - kms:Decrypt
```

Comment:
Just wondering if you also need `kms:GenerateDataKey*`, `ReEncrypt*` permissions too? Noticed you have them on the other PR, and I seem to remember there is some extra stuff required when KMS keys are used with SQS.

Reply:
Good point, I just copied this from the original role, but it's a good chance to see if we really need it, will try

Reply:
Have checked this now, and confirmed that we can do everything we need with reading from queues and stuff falling back to the DLQ with the permissions granted here.